client/src/main/java/org/spine3/base/FailureThrowable.java

```diff
@@ -38,22 +38,36 @@
 
     private static final long serialVersionUID = 0L;
 
-    // We accept GeneratedMessage (instead of Message) because
-    // generated messages implement Serializable.
+    /**
+     * For the {@code failureMessage} and {@code commandMessage} we accept GeneratedMessage
+     * (instead of Message) because generated messages implement Serializable.
+     */
     private final GeneratedMessageV3 failureMessage;
 
+    /**
+     * The message of the {@code Command}, which led to this {@code Failure}.
+     */
+    private final GeneratedMessageV3 commandMessage;
+
+    /**
+     * The context of the {@code Command}, which led to this {@code Failure}.
+     */
+    private final CommandContext commandContext;
+
     /** The moment of creation of this object. */
     private final Timestamp timestamp;
 
-    protected FailureThrowable(GeneratedMessageV3 failureMessage) {
+    protected FailureThrowable(GeneratedMessageV3 commandMessage,
+                               CommandContext ctx,
```

Comment:
Please rename the parameter to `commandContext` for consistency with the rest of the code.